src/extension.ts

```diff
@@ -14,9 +14,17 @@ import { Deferred } from './protocol';
 import * as utils from './models/utils';
 import { ObjectExplorerUtils } from './objectExplorer/objectExplorerUtils';
 import SqlToolsServerClient from './languageservice/serviceclient';
+import { ConnectionProfile } from './models/connectionProfile';
 
 let controller: MainController = undefined;
 
+class FirewallRuleError extends Error {
```

Comment:
This should go somewhere else - it's own file or generic utils file or something

Reply:
And actually - it would be good to add an interface for this in the extension typings file so that others using this can catch and handle these errors more easily.

Reply:
I already added interface. will add the "implements"